test/5.future-both.test.js

```diff
@@ -117,11 +69,51 @@ describe('FutureBoth', () => {
   describe('#toString()', () => {
 
     it('returns the code to create the FutureBoth', () => {
-      const m = Future.of(1).both(Future.of(2));
+      const m = both(Future.of(1), Future.of(2));
       const s = 'Future.of(1).both(Future.of(2))';
       expect(m.toString()).to.equal(s);
     });
 
   });
 
+};
+
+describe('Future.both()', () => {
+
+  it('is a curried binary function', () => {
+    expect(Future.both).to.be.a('function');
+    expect(Future.both.length).to.equal(2);
+    expect(Future.both(Future.of(1))).to.be.a('function');
+  });
+
+  it('throws when not given a Future as first argument', () => {
+    const f = () => Future.both(1);
+    expect(f).to.throw(TypeError, /Future.*first/);
+  });
+
+  it('throws when not given a Future as second argument', () => {
+    const f = () => Future.both(Future.of(1), 1);
+    expect(f).to.throw(TypeError, /Future.*second/);
+  });
+
+  //TODO: Argument order.
+  testInstance((a, b) => Future.both(a, b));
```

Comment:
Actually, it's quite expectable that `Future.both(ma, mb)` would give me `Future [a ,b]` and not `Future [b, a]`. I'm going to remove this comment.